src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -310,10 +314,13 @@ internal override sealed bool TryReadAsObject(ref Utf8JsonReader reader, JsonSer
         }
 
         /// <summary>
-        /// Overridden by the nullable converter to prevent boxing of values by the JIT.
+        /// Performance optimization. The 'in' modifier in TryWrite(in T Value) will cause boxing, so this helper method avoids that.
         /// </summary>
-        internal virtual bool IsNull(in T value) => value == null;
+        internal bool IsNull(T value) => value is null;
```

Comment:
```suggestion
        private bool IsNull(T value) => value is null;
```

Reply:
Also, this could now be a static local method inside `TryWrite` to avoid polluting the JsonConverter internal APIs.